client/X11/xf_client.c

```diff
@@ -475,17 +475,21 @@ BOOL xf_process_x_events(freerdp* instance)
 
 	while (pending_status)
 	{
-		xf_lock_x11(xfc, FALSE);
+		xf_lock_x11(xfc, TRUE);
 		pending_status = XPending(xfc->display);
-		xf_unlock_x11(xfc, FALSE);
+		xf_unlock_x11(xfc, TRUE);
 
 		if (pending_status)
 		{
+			xf_lock_x11(xfc, TRUE);
```

Comment:
Protect following X function.
We do correct thing in xf_input_thread, but seems forget here.